src/components/Sidebar/BranchTransition.tsx

```diff
@@ -27,12 +80,38 @@ const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, upda
         updatePathway={updatePathway}
         isTransition={true}
       />
+      {!useCriteriaSelected && (
+        <SidebarButton
+          buttonName="Use Criteria"
+          buttonIcon={<FontAwesomeIcon icon={faPlus} />}
+          buttonText="Add previously built or imported criteria logic to branch node."
+          onClick={handleUseCriteria}
+        />
+      )}
 
-      <SidebarButton
-        buttonName="Use Criteria"
-        buttonIcon={<FontAwesomeIcon icon={faPlus} />}
-        buttonText="Add previously built or imported criteria logic to branch node."
-      />
+      {useCriteriaSelected && (
+        <>
+          <DropDown
+            id="Criteria"
+            label="Criteria"
+            options={criteriaOptions}
+            onChange={selectCriteriaSource}
+            value={transition.condition?.cql || undefined}
+          />
+          <FormControl variant="outlined" fullWidth>
+            <TextField
```

Comment:
When I type into this textbox nothing happens. Unless is REALLY slow (like more than a minute) it's not working for me

Reply:
Weird. It works for me on Mac/Chrome. This comment should probably be added to the optimization task ([PATHWAYS-292](https://jira.mitre.org/browse/PATHWAYS-292)).